src/dist/replication/lib/log_block.h

```diff
@@ -76,5 +76,62 @@ class log_block
     void init();
 };
 
+// Append writes into a buffer which consists of one or more fixed-size log blocks.
+// Not thread-safe. Requires lock protection.
+class log_appender
+{
+public:
+    explicit log_appender(int64_t start_offset)
+    {
+        _blk = std::make_shared<log_block>(start_offset);
+    }
+
+    void append_mutation(const mutation_ptr &mu, const aio_task_ptr &cb);
+
+    // Consumes one log block from buffer.
+    // This function may not take all the pending writes when there are multiple blocks.
+    std::shared_ptr<log_block> take_log_block();
+
+    size_t size() const { return _full_blocks_size + (_blk ? _blk->size() : 0); }
+    size_t blob_count() const { return _full_blocks_blob_cnt + (_blk ? _blk->data().size() : 0); }
+    bool empty() const { return !_blk; }
+
+    std::vector<mutation_ptr> mutations() const;
```

Comment:
When we will get all mutations? I only see it is used in unit test.

Reply:
They're called during log learning, you can find the usages in `mutation_log_private::get_learn_state_in_memory`, and `get_in_memory_mutations`,

Replica learning requires both the checkpoint, logs on the disk, and buffered writes in memory, which is `mutations()` used for.